# Coding Exercises (Part 1)

Now, you will have the opportunity to practice what you have learned. <br>
__Follow the instructions__ and insert your code! 

The correct answer is provided below your coding cell. There you can check whether your code is correct.

If you need some help or if you want to check your code, you can also have a look at the __solutions__.

### Have Fun!

--------------------------------------------------------------------------------------------------------------

## Exercise 1: Using Python as a Calculator for TVM Problems

Use __variables__ where appropriate! __Input__ values and your __final results__ shall be stored in variables. 

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [2]:
# Store Input values in variables
pv=2000
n=7
r=0.04

In [3]:
# Calculate and store the future value in a variable
fv=pv*(1+r)**n

In [4]:
# print the future value
fv

2631.8635584716803

(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [45]:
# Store Input values in variables
fv=20000
n=4
r=0.035

In [46]:
# Calculate and store the present value in a variable
pv=fv/(1+r)**n

In [49]:
# print the present value
pv

17428.84455397145

(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [25]:
# Bank B

pv=10000
fv=15000
n=10
r=(((fv/pv)**0.1)-1)*100
r

4.137974399241062

In [24]:
# Bank A

pv=10000
n1=10
r1=0.045
fv1=pv*(1+r1)**n1
fv1

# I prefer the offer from Bank A

15529.694217328959

(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [26]:
# FV=PV*(1+r)^n
# r=((FV/PV)^1)-1
pv=976
fv=1053
n=1
r=(((fv/pv)**1)-1)*100
r

7.889344262295084

(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [27]:
pr=((fv/pv)-1)*100

In [28]:
pr

7.889344262295084

In [29]:
# Dividende (Yield) = 40USD

y=(40/pv)*100
y

4.098360655737705

In [31]:
# Gesamtrendite (Total Return):

tr=(((fv+y)/pv)-1)*100
tr

8.309258263907559

In [32]:
ergebnis=y+pr
ergebnis

11.987704918032788

(results: 4.10% and 11.99%)

In [33]:
print(pr,y,ergebnis,sep=', ')

7.889344262295084, 4.098360655737705, 11.987704918032788


6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

# Well Done!

---------------------------------------------------------------------------------------------------------------------

# Solutions (Stop here if you want to code on your own!)

# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [41]:
# Store Input values in variables
pv = 2000
n = 7
r = 0.04

In [42]:
# Calculate and store the future value in a variable
fv = pv * (1 + r)**n

In [43]:
# print the future value
fv

2631.8635584716803

(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [46]:
# Store Input values in variables
fv = 20000
n = 4
r = 0.035

In [47]:
# Calculate and store the present value in a variable
pv = fv / (1 + r)**n

In [48]:
# print the present value
pv

17428.84455397145

(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [49]:
# Store Input values in variables
pv = 10000
fv = 15000
n = 10

In [50]:
r = (fv / pv)**(1/n) - 1

In [51]:
r

0.04137974399241062

(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [52]:
p0 = 976
p1 = 1053
div = 40

In [53]:
p_ret = p1 / p0 - 1

In [54]:
p_ret

0.07889344262295084

(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [55]:
div_y = div / p0

In [56]:
div_y

0.040983606557377046

In [57]:
total_ret = p_ret + div_y

In [58]:
total_ret

0.11987704918032788

(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [59]:
print(p_ret, div_y, total_ret, sep = ", ")

0.07889344262295084, 0.040983606557377046, 0.11987704918032788
